### PR TITLE
convertible: use Document::YAML_FRONT_MATTER_REGEXP to parse transformable files

### PR DESCRIPTION
**Pull request by parkr**
Replaces #4736, which was my commit in a PR but with some git conflict.

/cc @jekyll/build

### CODE REVIEW

File: lib/jekyll/convertible.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def read_yaml(base, name, opts = {})
       begin
         self.content = File.read(site.in_source_dir(base, name),
                                  Utils.merged_file_read_opts(site, opts))
-        if content =~ /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
+        if content =~ Document::YAML_FRONT_MATTER_REGEXP
```
Comment (envygeeks) [inline lib/jekyll/convertible.rb]: This is only used in Convertible, why is on Document if it's only used here?

Comment (envygeeks) [inline lib/jekyll/convertible.rb]: Take that back.

Comment (parkr) [inline lib/jekyll/convertible.rb]: You're OK with it then?

Comment (envygeeks) [inline lib/jekyll/convertible.rb]: I see no problem.

**Comment by envygeeks** [comment]
:shipit:

**Comment by parkr** [comment]
😄 

@jekyllbot: merge +bug

**Pull request merged**